src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,256 @@
+(rmul_comm_rmul_rmul: ∀ a b : A, (b * a) * (a * a) = (b * (a * a)) * a)
+
+lemma jordan_operators (A : Type*) [non_unital_non_assoc_ring A] [jordan_ring A] (a : A) :
+(⁅L a, R a⁆ = 0) ∧ (⁅L a, L (a * a)⁆ = 0) ∧ (⁅L a, R (a * a)⁆ = 0) ∧ (⁅L (a * a), R a⁆ = 0) ∧
+(⁅R a, R (a * a)⁆ = 0) :=
+begin
+  rw [ring.lie_def, ring.lie_def, ring.lie_def, ring.lie_def, ring.lie_def ],
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_comm_rmul, rw sub_self, },
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_lmul_comm_lmul, rw sub_self, },
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_comm_rmul_rmul, rw sub_self, },
+  split,
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.lmul_lmul_comm_rmul, rw sub_self, },
+  { ext b,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      R_apply_apply, add_monoid.coe_mul, L_apply_apply],
+    rw jordan_ring.rmul_comm_rmul_rmul, rw sub_self, },
+end
+
+universe u
+
+/- A (unital, associative) ring satisfies the (non-commutative) Jordan axioms-/
+@[priority 100] -- see Note [lower instance priority]
+instance ring_jordan (B : Type u) [ring B] : jordan_ring (B) :=
+{ lmul_comm_rmul := by { intros, rw mul_assoc },
+  lmul_lmul_comm_lmul := by { intros, rw [mul_assoc, mul_assoc] },
+  lmul_comm_rmul_rmul := by { intros, rw [mul_assoc] },
+  lmul_lmul_comm_rmul := by { intros, rw [←mul_assoc] },
+  rmul_comm_rmul_rmul := by { intros, rw [← mul_assoc, ← mul_assoc] } }
+
+variables (A : Type*) [non_unital_non_assoc_ring A]
+
+/--
+A non unital, non-associative ring with a commutative Jordan multipication
+-/
+class comm_jordan :=
+(mul_comm: ∀ a b : A, a * b = b * a)
+(jordan: ∀ a b : A, (a * b) * (a * a) = a * (b * (a *a)))
+
+variable [comm_jordan A]
+
+/-
+instance : comm_monoid A :=
+{ mul_comm := λ a b, comm_jordan.mul_comm a b,
+  .. (show non_unital_non_assoc_ring A, by apply_instance) }
+-/
+
+lemma comm_jordan_operators  {A : Type*} [non_unital_non_assoc_ring A] [comm_jordan A] :
+  (∀ a: A, L a = R a) ∧ (∀ a: A, ⁅L a, L (a*a)⁆ = 0) :=
+begin
+  split,
+  { intro, ext b, simp, rw comm_jordan.mul_comm, },
+  { intro,
+    ext b,
+    rw ring.lie_def,
+    simp only [add_monoid_hom.zero_apply, add_monoid_hom.sub_apply, function.comp_app,
+      add_monoid.coe_mul, L_apply_apply],
+    rw [comm_jordan.mul_comm (a * a) (a * b), comm_jordan.jordan, comm_jordan.mul_comm b (a * a),
+    sub_self], }
+end
```

Comment:
This would be better as two lemmas

Reply:
Now I think about it, having shown that the commutative Jordan axioms imply the Jordan axioms, there's no need to restate these operator equations for the commutative case, so I've removed this lemma altogether. Thanks.